### PR TITLE
task owner: remove remains of obsolete functionality

### PR DESCRIPTION
**Pull request by oliver-sanders**
follow on from: #3990
closes: #3593

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why?).
- [x] No change log entry required (functionality already removed).
- [x] No documentation update required (document as part of larger changes).
- [x] No dependency changes.

### CODE REVIEW

File: cylc/flow/data_messages.proto [hunk 1]
```diff
@@ -117,7 +117,6 @@ message PbJob {
     optional string host = 15;
     optional string init_script = 16;
     optional string job_log_dir = 17;
-    optional string owner = 18;
```
Comment (oliver-sanders) [inline cylc/flow/data_messages.proto:120]: @dwsutherland good to remove like this?

**Review by wxtim — APPROVED** [review]
- [x] Seems rational.
- [x] Doesn't break any tests.
- [ ] @dwsutherland  okayed changes to `data_messages.proto`.

Approved, conditional on that last.

File: cylc/flow/data_messages.proto (hunk 1 above)
Comment (kinow) [reply cylc/flow/data_messages.proto:120]: I think after the 8.0 final release it shouldn't happen. Instead we should probably use

```
optional string owner = 18 [deprecated=true];  // No longer used, see issue-1234; to be removed in 8.1 or 9.0...
```

But I think it should be OK to remove it now, but @dwsutherland is the best to answer that. Also, David, if this is removed now, do we re-organize the other field numbers (i.e. re-assign the `18` to the `19`, and so on?)?

**Review by kinow — APPROVED** [review]
Looks good to me. Pending @dwsutherland to check the review questions :+1:

**Comment by hjoliver** [comment]
Ping @dwsutherland - small question for you, to finish this one off (when you're ready).

File: cylc/flow/data_messages.proto (hunk 1 above)
Comment (dwsutherland) [reply cylc/flow/data_messages.proto:120]: Yeah, that's fine.. Number tag doesn't need to be consecutive

**Review by dwsutherland — APPROVED** [review]
Looks good to me! (sorry for the late review)

You'll need to rebase, and the `job_pool.py` is now merged into `data_store_mgr.py` .. (And I'm sure you know; it's always safer to regenerate the `data_messages_pb2.py` then to manually resolve)

👍

**Comment by oliver-sanders** [comment]
Rebased @wxtim could you sanity check and merge.

**Pull request merged**